filecoin/retriever.go

```diff
@@ -113,6 +119,9 @@ func NewRetriever(
 			suspensionDuration:       time.Minute,
 			failureHistoryDuration:   time.Second * 15,
 		}),
+		confirm: func(c cid.Cid) (bool, error) {
```

Comment:
Why not just have the parameter be the confirm signature function and then pass blockManager.Has
(i.e. the interface looks superfluos)

Reply:
yeah, ok, but `ctx` is also a problem here, I can push that back up to autoretrieve though